tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+            Err("Receiver is not currently a consumer - must have 'running' status!")
+        } else {
+            // update the running validator set, but not the history yet,
+            // as that only happens when the next block is started
+            val currentConsumerState = currentState.consumerStates.get(receiver)
+            val newConsumerState = currentConsumerState.with(
+                "chainState",
+                currentConsumerState.chainState.with(
+                    "currentValidatorSet", packet.validatorSet
+                )
+            ).with(
+                "maturationTimes",
+                currentConsumerState.maturationTimes.put(
+                    packet, 
+                    currentConsumerState.chainState.lastTimestamp + UnbondingPeriodPerChain.get(receiver)
+                )
+            ).with(
+                "receivedVSCPackets",
+                currentConsumerState.receivedVSCPackets.prepend(packet)
+            )
+            val newConsumerStates = currentState.consumerStates.set(receiver, newConsumerState)
+            val newState = currentState.with(
+                "consumerStates", newConsumerStates
+            )
+            Ok(newState)
+        }
+    }
+
+    // receives a given packet on the provider. The arguments are the consumer chain that sent the packet, and the packet itself.
+    // To receive a packet, add it to the list of received maturations.
+    pure def recvPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: VSCMaturedPacket): Result = {
+        if (not(isCurrentlyConsumer(sender, currentState.providerState))) {
+            Err("Sender is not currently a consumer - must have 'running' status!")
+        } else if (currentState.providerState.sentVSCPackets.get(sender).head().id != packet.id) {
+            // the packet is not the oldest sentVSCPacket, something went wrong
+            Err("Received maturation is not for the oldest sentVSCPacket")
+        } else {
+            val currentReceivedMaturations = currentState.providerState.receivedMaturations
+            val newReceivedMaturations = currentReceivedMaturations.union(Set(packet))
+            val newProviderState = currentState.providerState.with(
+                "receivedMaturations", newReceivedMaturations
+            )
+            // prune the sentVSCPacket
+            val newSentVSCPacket = currentState.providerState.sentVSCPackets.get(sender).tail()
+            val newState = currentState.with(
+                "providerState", newProviderState
+            )
+            Ok(newState)
+        }
+    }
+
+    // removes the oldest outstanding packet from the consumer. on-chain, this would happen when the packet is acknowledged.
+    // only the oldest packet can be removed, since we model ordered channels.
+    pure def removeOutstandingPacketFromConsumer(currentState: ProtocolState, sender: Chain): Result = {
```

Comment:
Because `removeOutstandingPacketFromConsumer` never returns an error, does it make sense to just have it return the `Result's underlying `ProtocolState`?

Reply:
True